daemon/indexing/listener/listener.js

```diff
@@ -182,9 +174,40 @@ async function runBatch(opts, context) {
   return lastLogBlock
 }
 
+async function withRetrys(fn) {
```

Comment:
nit: Add a comment explaining this retries up to N times, with exponential backoff.